File: be/src/project-members/dto/update-project-member.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsEnum } from 'class-validator';

export class UpdateProjectMemberDto {
  @ApiProperty({ example: 'leader' })
  @IsEnum(['leader', 'editor', 'viewer'])
  role: 'leader' | 'editor' | 'viewer';
}

File: backend-java/src/main/java/com/qlda/backend/projectmember/dto/UpdateProjectMemberDto.java
package com.qlda.backend.projectmember.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

@Data
public class UpdateProjectMemberDto {
    @NotBlank(message = "Role không được để trống")
    @Pattern(regexp = "leader|editor|viewer", message = "Role phải là leader, editor hoặc viewer")
    private String role;
}

